dali/pipeline/operators/reader/loader/video_loader.h

```diff
@@ -116,6 +133,9 @@ class VideoLoader : public Loader<GPUBackend, SequenceWrapper> {
       stop_(false) {
     if (step_ < 0)
       step_ = count_ * stride_;
+
+    file_label_pair_ = filesystem::get_file_label_pair(file_root_, filenames_);
```

Comment:
Please move this call to `PrepareMetadataImpl`

Reply:
In the case that `filenames` argument is passed, at least one of the filenames is already known and this is used to fetch height, width using `VideoLoader::load_width_height()`. 

If we move this to `PrepareMetadataImpl` and `file_root` is passed, then the `load_width_heigh()` call below will fail because the `file_root` has not been crawled yet and no filenames are known.

Do you have any suggestions how to avoid this ?

https://github.com/NVIDIA/DALI/blob/1af6beb372e457f9c0c7436adec375ab6441eb58/dali/pipeline/operators/reader/video_reader_op.h#L48-L55

Reply:
Oh, right... the whole lazy_init currently doesn't work with the VideoReader actually.

First thing to do would be here is to move the whole VideoReader constructor body to a function that would be std::call_once'd in RunImpl (but that's kind of out of the scope of this PR).

Leave it as it is for now